test/MorphoTokenOptimism.t.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Test, console} from "lib/forge-std/src/Test.sol";
+import {MorphoTokenOptimism} from "../src/MorphoTokenOptimism.sol";
+import {ERC1967Proxy} from
+    "lib/openzeppelin-contracts-upgradeable/lib/openzeppelin-contracts/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {UUPSUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+
+contract MorphoTokenOptimismTest is Test {
+    address internal constant MORPHO_DAO = 0xcBa28b38103307Ec8dA98377ffF9816C164f9AFa;
+    address internal REMOTE_TOKEN;
+    address internal BRIDGE;
+
+    MorphoTokenOptimism public tokenImplem;
+    MorphoTokenOptimism public morphoOptimism;
+    ERC1967Proxy public tokenProxy;
+
+    uint256 internal constant MIN_TEST_AMOUNT = 100;
+    uint256 internal constant MAX_TEST_AMOUNT = 1e28;
+
+    function setUp() public virtual {
+        REMOTE_TOKEN = makeAddr("RemoteToken");
+        BRIDGE = makeAddr("Bridge");
+
+        // DEPLOYMENTS
+        tokenImplem = new MorphoTokenOptimism();
+        tokenProxy = new ERC1967Proxy(address(tokenImplem), hex"");
+
+        morphoOptimism = MorphoTokenOptimism(payable(address(tokenProxy)));
+        morphoOptimism.initialize(MORPHO_DAO, REMOTE_TOKEN, BRIDGE);
+    }
+
+    function testInitilizeZeroAddress(address randomAddress) public {
+        vm.assume(randomAddress != address(0));
+
+        address proxy = address(new ERC1967Proxy(address(tokenImplem), hex""));
+
+        vm.expectRevert();
+        MorphoTokenOptimism(proxy).initialize(address(0), randomAddress, randomAddress);
+
+        vm.expectRevert();
+        MorphoTokenOptimism(proxy).initialize(randomAddress, address(0), randomAddress);
+
+        vm.expectRevert();
+        MorphoTokenOptimism(proxy).initialize(randomAddress, randomAddress, address(0));
+    }
+
+    function testUpgradeNotOwner(address updater) public {
+        vm.assume(updater != address(0));
+        vm.assume(updater != MORPHO_DAO);
+
+        address newImplem = address(new MorphoTokenOptimism());
+
+        vm.expectRevert();
+        morphoOptimism.upgradeToAndCall(newImplem, hex"");
+    }
+
+    function testUpgrade() public {
+        address newImplem = address(new MorphoTokenOptimism());
+
+        vm.prank(MORPHO_DAO);
+        morphoOptimism.upgradeToAndCall(newImplem, hex"");
+    }
+
+    function testGetters() public view {
+        assertEq(morphoOptimism.remoteToken(), REMOTE_TOKEN, "remoteToken");
+        assertEq(morphoOptimism.bridge(), BRIDGE, "bridge");
+    }
+
+    function testMintNoBridge(address account, address to, uint256 amount) public {
+        vm.assume(account != address(0));
+        vm.assume(to != address(0));
+        vm.assume(account != BRIDGE);
+        amount = bound(amount, MIN_TEST_AMOUNT, MAX_TEST_AMOUNT);
+
+        vm.expectRevert();
```

Comment:
I think we should expect the correct error message

Reply:
As the `require` takes a custom error as input, I don't know how to check this

I tried `vm.expectRevert(Error's selector)` but it doesn't seem to work

<img width="1147" alt="Capture d’écran 2024-10-18 à 17 10 07" src="https://github.com/user-attachments/assets/db731dc8-cb79-4a13-b675-73b4686d5634">